docs/source/plugins/for_plugin_developers.rst

```diff
@@ -0,0 +1,274 @@
+   from pluggy import HookimplMarker
+
+   napari_hook_implementation = HookimplMarker("napari")
+
+
+Currently (as of March, 2020), the only way that napari knows *which* hook
+specification your implementation matches is by looking at the *name* of your
+function.  So in the example above, it was critical that our hook
+implementation was literally named ``napar_get_reader``:
+
+
+.. code-block:: python
+
+   @napari_hook_implementation
+   def napari_get_reader(path: str):
+      ...
+
+However, `a pull request has been merged at pluggy
+<https://github.com/pytest-dev/pluggy/pull/251>`_ that will enable you to mark
+*any* function as satisfying a napari hook specification (regardless of the
+function's name) using the following syntax:
+
+.. code-block:: python
+
+   @napari_hook_implementation(specname="napari_get_reader")
+   def whatever_name_you_want(path: str):
+      ...
+
+(Monitor the `pluggy changelog
+<https://github.com/pytest-dev/pluggy/blob/master/CHANGELOG.rst>`_ for release
+of PR #251.)
+
+.. _plugin-discover:
+
+Step 3: Make your plugin discoverable
+-------------------------------------
+
+Packages and modules installed in the same environment as ``napari`` may make
+themselves "discoverable" to napari using one of two conventions:
+
+Using naming convention
+^^^^^^^^^^^^^^^^^^^^^^^
+
+``napari`` will look for *hook implementations* (i.e. functions decorated with
+the ``HookimplMarker("napari")`` decorator) in all top-level modules in
+``sys.path`` that begin with the name ``napari_`` (e.g. "``napari_myplugin``").
+
+One potential benefit of using discovery by naming convention is that it will
+allow ``napari`` to query the PyPi API to search for potential plugins.
+
+Using package metadata
+^^^^^^^^^^^^^^^^^^^^^^
+
+By providing an ``entry_points`` argument with the key ``napari.plugin`` to
+``setup()`` in ``setup.py``, plugins can register themselves for discovery
+(even if their names do not begin with "``napari_``").
+
+For example if you have a package named ``mypackage`` with a submodule
+``napari_plugin`` where you have decorated one or more napari hook
+implementations, then if you include in ``setup.py``:
+
+.. code-block:: python
+
+   # setup.py
+
+   setup(
+      ...
+      entry_points={'napari.plugin': 'plugin_name = mypackage.napari_plugin'},
+      ...
+   )
+
+... then napari will search the ``mypackage.napari_plugin`` module for
+functions decorated with the ``HookimplMarker("napari")`` decorator and
+register them the plugin name ``"plugin_name"``.
+
+One benefit of using this approach is that if you already have an existing
+pip-installable package, you can extend support for ``napari`` without having
+to rename your package, simply by identifying the module in your package that
+has the hook implementations.
+
+A user would then be able to use ``napari``, extended with your package's
+functionality by simply installing your package along with napari:
+
+.. code:: bash
+
+   pip install napari mypackage
```

Comment:
I feel like this needs a follow-up line:
```suggestion
   pip install napari mypackage

which is 🎵 ah-maziiing! 🎵 
```

😀

Reply:
umm ..... not so sure about this one ;-)

Reply:
but it is amazing :-)

Reply:
maybe just "which is amazing!"